### PR TITLE
Report UseSite diagnostics on user-defined conversions

### PR DESCRIPTION
**Pull request by 333fred**
We weren't reporting use site diagnostics on user-defined conversion operators, meaning that we would miss things like UnmanagedCallersOnly or invalid modreq reporting. Fixes https://github.com/dotnet/roslyn/issues/54113.

### CODE REVIEW

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for review.

File: src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs:93]: >if (conversion.Method is not null) [](http://example.com/codeflow?start=12&length=34)

There are many other places where we create Conversion objects like this, but do not pass them to this method. I am planning to remedy that in 17.1.  So, I think it is fine limit the fix to this code path.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConversionTests.cs
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConversionTests.cs:1220]: >DisallowedModreqOnConversion [](http://example.com/codeflow?start=20&length=28)

This looks like dupe of the other test that was already added to the other test file.

Comment (333fred) [reply src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConversionTests.cs:1220]: I wanted to have a test that covered a modreq, as opposed to covering `UnmanagedCallersOnly`. Previous to this change, this test would not have reported any errors.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1), modulo the duplicate test.

File: src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConversionTests.cs
Comment (AlekseyTs) [reply src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConversionTests.cs:1220]: > I wanted to have a test that covered a modreq

Sorry, missed that.

**Comment by AlekseyTs** [comment]
>             // Conversions aren't flagged due to https://github.com/dotnet/roslyn/issues/54113.

Remove?

---
Refers to: src/Compilers/CSharp/Test/Symbol/Symbols/StaticAbstractMembersInInterfacesTests.cs:26654 in c1db171. [](commit_id = c1db171152aed6f9c07fab018718976aeeeefcd8, deletion_comment = False)

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Comment by 333fred** [comment]
@dotnet/roslyn-compiler for a second review.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 3)

**Pull request merged**
